src/libraries/System.Private.CoreLib/src/System/Char.cs

```diff
@@ -225,6 +225,76 @@ public static bool TryParse([NotNullWhen(true)] string? s, out char result)
         //
         // Static Methods
         //
+
+        /// <summary>Indicates whether a character is categorized as an ASCII letter.</summary>
+        /// <param name="c">The character to evaluate.</param>
+        /// <returns>true if <paramref name="c"/> is an ASCII letter; otherwise, false.</returns>
+        /// <remarks>
+        /// This determines whether the character is in the range 'A' through 'Z', inclusive,
+        /// or 'a' through 'z', inclusive.
+        /// </remarks>
+        public static bool IsAsciiLetter(char c) => (uint)((c | 0x20) - 'a') <= 'z' - 'a';
```

Comment:
Open question: is performing the SUB before the OR preferred because then the move can be folded into a single LEA operation? (I honestly don't know.)